src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1222,486 +1631,614 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
 
-    if ((this->unk_390 & 0x1000) && (((actor != NULL) && (actor->id == ACTOR_OBJ_AQUA) && (actor->params & 1)) ||
-                                     (this->actor.colChkInfo.damageEffect == 2))) {
-        this->actionFunc = func_80A14E74;
+    if ((this->actionFlags & ENGO_FLAG_HIT_BY_OTHER) &&
+        (((actorCollidedWith != NULL) && (actorCollidedWith->id == ACTOR_OBJ_AQUA) &&
+          (actorCollidedWith->params & 1)) ||
+         (this->actor.colChkInfo.damageEffect == ENGO_DMGEFF_FIRE))) {
+        this->actionFunc = EnGo_AwaitThaw;
     }
 }
 
-void func_80A14E74(EnGo* this, PlayState* play) {
-    if (func_80A134F4(this, this->actor.cutscene)) {
-        this->actionFunc = func_80A14EB0;
+/**
+ * Action Function for transitioning to thawing
+ */
+void EnGo_AwaitThaw(EnGo* this, PlayState* play) {
+    if (EnGo_CheckAndChangeCutscene(this, this->actor.cutscene)) {
+        this->actionFunc = EnGo_Thaw;
     }
 }
 
-void func_80A14EB0(EnGo* this, PlayState* play) {
-    EnGo* sp24 = (EnGo*)this->actor.child;
+/**
+ * Action function for thawing of the goron, melting sounds, and steam.
+ */
+void EnGo_Thaw(EnGo* this, PlayState* play) {
+    EnGo* unfrozenBro = (EnGo*)this->actor.child;
 
-    if ((s32)(this->unk_39C * 3.0f) != 0) {
+    if ((s32)(this->iceBlockScale * 3.0f) != 0) {
         Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_ICE_MELT_LEVEL - SFX_FLAG);
-        Math_ApproachF(&this->unk_39C, 0.0f, 0.02f, 1.0f);
-        this->unk_3A0 = (this->unk_39C / 0.9f) * 100.0f;
-        func_80A139E4(this);
+        Math_ApproachF(&this->iceBlockScale, 0.0f, 0.02f, 1.0f);
+        this->iceBlockAlpha = (this->iceBlockScale / 0.9f) * 100.0f;
+        EnGo_MakeSteam(this);
     } else {
         ActorCutscene_Stop(this->actor.cutscene);
-        func_80A143A8(this, play);
-        if ((ENGO_GET_F(&this->actor) == ENGO_F_4) && (ENGO_GET_70(&this->actor) == ENGO_70_1)) {
-            SubS_UpdateFlags(&this->unk_390, 4, 7);
-            func_80A143A8(sp24, play);
-            sp24->actionFunc = func_80A149B0;
+        EnGo_UpdateToShiveringAnimation(this, play);
+        if ((ENGO_GET_TYPE(&this->actor) == ENGO_GRAVEBRO) &&
+            (ENGO_GET_SUBTYPE(&this->actor) == ENGO_GRAVEBRO_FROZEN)) {
+            SubS_UpdateFlags(&this->actionFlags, 4, 7);
+            EnGo_UpdateToShiveringAnimation(unfrozenBro, play);
+            unfrozenBro->actionFunc = EnGo_Idle;
         }
-        this->actionFunc = func_80A149B0;
+        this->actionFunc = EnGo_Idle;
     }
 }
 
-void func_80A14FC8(EnGo* this, PlayState* play) {
-    s32 sp38[] = {
-        0, 2, 6, 20, 18, 5, 5, 15,
+/**
+ * Cutscene action function for the arrival of springtime
+ *
+ * This cutscene consists of:
+ * - The non-frozen brother stops shivering and looks around.
+ * - The frozen brother thaws and shivers briefly.
+ * - The non-frozen brother pats his brother on the back and shows him the change in temperature.
+ * - They both look around
+ * - They both cheer.
```

Comment:
I don't think I have the background on the cutscene system to figure figure out quite how this operated, as my initial interpretation is off since the csAction numbers aren't chronological (i.e. springArrivalCurrentCsAction of 6 is the melting of the frozen goron, but 4 is where one actor would be interacting with the already thawed goron. '2' is missing entirely... It seems like maybe each actor moves through the actions independently? Anyways, if there's background details on that that could help me name the csAction's properly, please point me in the general area and I'll try to figure it out.

Reply:
I suppose it can wait until we get this sort of thing documented.